### PR TITLE
fix: correct username and token validation

### PR DESCRIPTION
**Pull request by Trott**
The current username and token validation regular expressions will match
any string. This adds tests and fixes the regular expressions.

### CODE REVIEW

File: lib/auth.js [hunk 1]
```diff
@@ -11,9 +10,24 @@ const ghauth = util.promisify(ghauthBase);
...
+  process.exit(1);
+}
+
 function check(username, token) {
-  assert(typeof username === 'string' && /^[a-zA-Z0-9]*/.test(username));
-  assert(typeof token === 'string' && /^[0-9a-f]*/.test(token));
+  if (typeof username !== 'string') {
+    errorExit(`username must be a string, received ${typeof username}`);
+  }
+  if (!/^[a-zA-Z0-9]+$/.test(username)) {
```
Comment (richardlau) [inline lib/auth.js:22]: These are GitHub usernames, aren't they? They're allowed to contain hyphens, at least according to https://github.com/join:
image

Comment (Trott) [inline lib/auth.js:22]:
```suggestion
  if (!/^[a-zA-Z0-9\-]+$/.test(username)) {
```

Comment (Trott) [reply lib/auth.js]: I added hyphens as a permissible character.

File: lib/auth.js [hunk 2]
```diff
@@ -11,9 +10,24 @@ const ghauth = util.promisify(ghauthBase);
...
+  process.exit(1);
+}
+
 function check(username, token) {
-  assert(typeof username === 'string' && /^[a-zA-Z0-9]*/.test(username));
-  assert(typeof token === 'string' && /^[0-9a-f]*/.test(token));
+  if (typeof username !== 'string') {
+    errorExit(`username must be a string, received ${typeof username}`);
+  }
+  if (!/^[a-zA-Z0-9\-]+$/.test(username)) {
```
Comment (Trott) [inline lib/auth.js:22]:
```suggestion
  if (!/^[a-zA-Z0-9-]+$/.test(username)) {
```

File: lib/auth.js [hunk 3]
```diff
@@ -11,9 +10,24 @@ const ghauth = util.promisify(ghauthBase);
...
+}
+
 function check(username, token) {
-  assert(typeof username === 'string' && /^[a-zA-Z0-9]*/.test(username));
-  assert(typeof token === 'string' && /^[0-9a-f]*/.test(token));
+  if (typeof username !== 'string') {
+    errorExit(`username must be a string, received ${typeof username}`);
+  }
+  if (!/^[a-zA-Z0-9-]+$/.test(username)) {
+    errorExit(`username must be alphanumeric, received ${username}`);
```
Comment (richardlau) [inline lib/auth.js:23]: Something more in line with GitHub error text (without the bits we're not actually checking for)?
```suggestion
    errorExit(`username may only contain alphanumeric characters or hyphens, received ${username}`);
```

**Review by richardlau — APPROVED** [review]

**Pull request merged**